beard/clustering/wrappers.py

```diff
@@ -151,8 +170,26 @@ def fit(self, X, y=None):
                                       criterion=self.criterion,
                                       depth=self.depth, R=self.R,
                                       monocrit=self.monocrit)
-
-                score = self.scoring(y[train], labels[train])
+                if ground_truth:
+                    train = (y != -1)
+
+                    if not self.affinity_score:
+                        score = self.scoring(y[train], labels[train])
+                    else:
+                        # For scoring function that accepts affinity
+                        score = self.scoring(y[train], labels[train], X)
+
+                elif self.affinity_score:
+                    n_labels = len(np.unique(labels))
+                    n_samples = Xs.shape[0]
+
+                    if 1 < n_labels < n_samples:
+                        score = self.scoring(Xs, labels)
+                    else:
+                        score = -np.inf
+                else:
+                    # could have a scoring function that accepts only labels
+                    return self
```

Comment:
> The test cases I reach this error with is when we specify the number of clusters or the treshold. I would need to change the logic of handeling these 2 cases in order to make this change. Is it fine to do that?

Yes, as long as the output remains the same and we can keep doing what was allowed before, it is fine with me.